convert2rhel/applock.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright(C) 2023 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import errno
+import io
+import logging
+import os
+import tempfile
+
+
+_DEFAULT_LOCK_DIR = "/var/lock/run"
+loggerinst = logging.getLogger(__name__)
```

Comment:
Rodolfo and I discussed this--I ended up doing the same thing that require_root() does, except I write to stderr. This could all stand a cleanup in the future.

Reply:
Note, I mentioned this because of the use of logging in ApplicationLock.  Since most of the code in ApplicationLock will also be used before `convert2rhel.main.initialize_logger()` is called, those will also be using logger library defaults instead of our customizations.